File: fontes-estruturado/principal/loopback/src/main/java/br/com/gersis/loopback/modelo/CampanhaAdsTeste.java
package br.com.gersis.loopback.modelo;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.strongloop.android.loopback.Model;
import com.strongloop.android.remoting.BeanUtil;
import org.json.JSONObject;


public class CampanhaAdsTeste extends Model {


	private String nome;
	private int prontaParaTeste;
	private String codigoAds;
	private String dataEnvioGoogle;
	private double custoTotal;
	private double receitaTotal;
	private String dataInicio;
	private String dataTermino;
	private int qtdeImpressao;
	private int qtdeClique;
	private double ctr;
	private String nomeAds;
	private double cpc;
	private int ativa;
	private double cpaMax;
	private double cpcMax;
	private String codigoAdsGrupoAnuncio;
	private String codigoAdsCampanha;
	private String primaryStatus;
	private String primaryStatusReasons;
	// Relacionamentos 1
	private ModeloCampanhaAdsTeste ModeloCampanhaAdsTeste;
	private ProdutoAfiliadoHotmart ProdutoAfiliadoHotmart;
	private ContaGoogle ContaGoogle;
	private PaginaVenda PaginaVenda;
	// Relacionamentos N
	private List<AnuncioCampanhaAdsTeste> AnuncioCampanhaAdsTestes;
	private List<PalavraChaveCampanhaAdsTeste> PalavraChaveCampanhaAdsTestes;
	private List<CampanhaAdsMetrica> CampanhaAdsMetricas;
	private List<CampanhaAdsMetricaIntraday> CampanhaAdsMetricaIntradays;

	public void setId(Long id) {
		this.setIdObjeto(id);
	}
	public void setId(Integer id) {
		this.setIdObjeto(id);
	}

	public int getIdInteger() {
		return new Integer(getId().toString());
	}
	public long getIdLong() {
		return new Long(getId().toString());
	}

	public JSONObject getJSON() {
		JSONObject obj = new JSONObject();
		try {
			obj.put("id",getId());
			obj.put("nome", nome);
			obj.put("prontaParaTeste", prontaParaTeste);
			obj.put("codigoAds", codigoAds);
			obj.put("dataEnvioGoogle", dataEnvioGoogle);
			obj.put("custoTotal", custoTotal);
			obj.put("receitaTotal", receitaTotal);
			obj.put("dataInicio", dataInicio);
			obj.put("dataTermino", dataTermino);
			obj.put("qtdeImpressao", qtdeImpressao);
			obj.put("qtdeClique", qtdeClique);
			obj.put("ctr", ctr);
			obj.put("nomeAds", nomeAds);
			obj.put("cpc", cpc);
			obj.put("ativa", ativa);
			obj.put("cpaMax", cpaMax);
			obj.put("cpcMax", cpcMax);
			obj.put("codigoAdsGrupoAnuncio", codigoAdsGrupoAnuncio);
			obj.put("codigoAdsCampanha", codigoAdsCampanha);
			obj.put("primaryStatus", primaryStatus);
			obj.put("primaryStatusReasons", primaryStatusReasons);
		} catch (Exception e) {
			e.printStackTrace();
		}
		return obj;
	}


	public void setNome(String valor) { 
		this.nome = valor;
	}
	public String getNome() { 
		return this.nome;
	}
	public void setProntaParaTeste(int valor) { 
		this.prontaParaTeste = valor;
	}
	public int getProntaParaTeste() { 
		return this.prontaParaTeste;
	}
	public void setCodigoAds(String valor) { 
		this.codigoAds = valor;
	}
	public String getCodigoAds() { 
		return this.codigoAds;
	}
	public void setDataEnvioGoogle(String valor) { 
		this.dataEnvioGoogle = valor;
	}
	public String getDataEnvioGoogle() { 
		return this.dataEnvioGoogle;
	}
	public void setCustoTotal(double valor) { 
		this.custoTotal = valor;
	}
	public double getCustoTotal() { 
		return this.custoTotal;
	}
	public void setReceitaTotal(double valor) { 
		this.receitaTotal = valor;
	}
	public double getReceitaTotal() { 
		return this.receitaTotal;
	}
	public void setDataInicio(String valor) { 
		this.dataInicio = valor;
	}
	public String getDataInicio() { 
		return this.dataInicio;
	}
	public void setDataTermino(String valor) { 
		this.dataTermino = valor;
	}
	public String getDataTermino() { 
		return this.dataTermino;
	}
	public void setQtdeImpressao(int valor) { 
		this.qtdeImpressao = valor;
	}
	public int getQtdeImpressao() { 
		return this.qtdeImpressao;
	}
	public void setQtdeClique(int valor) { 
		this.qtdeClique = valor;
	}
	public int getQtdeClique() { 
		return this.qtdeClique;
	}
	public void setCtr(double valor) { 
		this.ctr = valor;
	}
	public double getCtr() { 
		return this.ctr;
	}
	public void setNomeAds(String valor) { 
		this.nomeAds = valor;
	}
	public String getNomeAds() { 
		return this.nomeAds;
	}
	public void setCpc(double valor) { 
		this.cpc = valor;
	}
	public double getCpc() { 
		return this.cpc;
	}
	public void setAtiva(int valor) { 
		this.ativa = valor;
	}
	public int getAtiva() { 
		return this.ativa;
	}
	public void setCpaMax(double valor) { 
		this.cpaMax = valor;
	}
	public double getCpaMax() { 
		return this.cpaMax;
	}
	public void setCpcMax(double valor) { 
		this.cpcMax = valor;
	}
	public double getCpcMax() { 
		return this.cpcMax;
	}
	public void setCodigoAdsGrupoAnuncio(String valor) { 
		this.codigoAdsGrupoAnuncio = valor;
	}
	public String getCodigoAdsGrupoAnuncio() { 
		return this.codigoAdsGrupoAnuncio;
	}
	public void setCodigoAdsCampanha(String valor) { 
		this.codigoAdsCampanha = valor;
	}
	public String getCodigoAdsCampanha() { 
		return this.codigoAdsCampanha;
	}
	public void setPrimaryStatus(String valor) { 
		this.primaryStatus = valor;
	}
	public String getPrimaryStatus() { 
		return this.primaryStatus;
	}
	public void setPrimaryStatusReasons(String valor) { 
		this.primaryStatusReasons = valor;
	}
	public String getPrimaryStatusReasons() { 
		return this.primaryStatusReasons;
	}

	public ModeloCampanhaAdsTeste getModeloCampanhaAdsTeste() {
		return ModeloCampanhaAdsTeste;
	}
	public void setModeloCampanhaAdsTeste(HashMap valor) {
		this.ModeloCampanhaAdsTeste = new ModeloCampanhaAdsTeste();
		BeanUtil.setProperties(this.ModeloCampanhaAdsTeste, (Map<String, ? extends Object>) valor, true);
	}
	public ProdutoAfiliadoHotmart getProdutoAfiliadoHotmart() {
		return ProdutoAfiliadoHotmart;
	}
	public void setProdutoAfiliadoHotmart(HashMap valor) {
		this.ProdutoAfiliadoHotmart = new ProdutoAfiliadoHotmart();
		BeanUtil.setProperties(this.ProdutoAfiliadoHotmart, (Map<String, ? extends Object>) valor, true);
	}
	public ContaGoogle getContaGoogle() {
		return ContaGoogle;
	}
	public void setContaGoogle(HashMap valor) {
		this.ContaGoogle = new ContaGoogle();
		BeanUtil.setProperties(this.ContaGoogle, (Map<String, ? extends Object>) valor, true);
	}
	public PaginaVenda getPaginaVenda() {
		return PaginaVenda;
	}
	public void setPaginaVenda(HashMap valor) {
		this.PaginaVenda = new PaginaVenda();
		BeanUtil.setProperties(this.PaginaVenda, (Map<String, ? extends Object>) valor, true);
	}
	public List<AnuncioCampanhaAdsTeste> getAnuncioCampanhaAdsTestes() {
		return  AnuncioCampanhaAdsTestes;
	}
	public void setAnuncioCampanhaAdsTestes(List<AnuncioCampanhaAdsTeste> valores) {
		this.AnuncioCampanhaAdsTestes = new ArrayList<AnuncioCampanhaAdsTeste>();
		for (int i = 0; i < valores.size(); i++) {
			Object objeto = new AnuncioCampanhaAdsTeste();
			System.out.println(" --> ObjetoMap ");
			BeanUtil.setProperties(objeto, (Map<String, ? extends Object>) valores.get(i), true);
			this.AnuncioCampanhaAdsTestes.add((AnuncioCampanhaAdsTeste) objeto);
		}
	}
	public List<PalavraChaveCampanhaAdsTeste> getPalavraChaveCampanhaAdsTestes() {
		return  PalavraChaveCampanhaAdsTestes;
	}
	public void setPalavraChaveCampanhaAdsTestes(List<PalavraChaveCampanhaAdsTeste> valores) {
		this.PalavraChaveCampanhaAdsTestes = new ArrayList<PalavraChaveCampanhaAdsTeste>();
		for (int i = 0; i < valores.size(); i++) {
			Object objeto = new PalavraChaveCampanhaAdsTeste();
			System.out.println(" --> ObjetoMap ");
			BeanUtil.setProperties(objeto, (Map<String, ? extends Object>) valores.get(i), true);
			this.PalavraChaveCampanhaAdsTestes.add((PalavraChaveCampanhaAdsTeste) objeto);
		}
	}
	public List<CampanhaAdsMetrica> getCampanhaAdsMetricas() {
		return  CampanhaAdsMetricas;
	}
	public void setCampanhaAdsMetricas(List<CampanhaAdsMetrica> valores) {
		this.CampanhaAdsMetricas = new ArrayList<CampanhaAdsMetrica>();
		for (int i = 0; i < valores.size(); i++) {
			Object objeto = new CampanhaAdsMetrica();
			System.out.println(" --> ObjetoMap ");
			BeanUtil.setProperties(objeto, (Map<String, ? extends Object>) valores.get(i), true);
			this.CampanhaAdsMetricas.add((CampanhaAdsMetrica) objeto);
		}
	}
	public List<CampanhaAdsMetricaIntraday> getCampanhaAdsMetricaIntradays() {
		return  CampanhaAdsMetricaIntradays;
	}
	public void setCampanhaAdsMetricaIntradays(List<CampanhaAdsMetricaIntraday> valores) {
		this.CampanhaAdsMetricaIntradays = new ArrayList<CampanhaAdsMetricaIntraday>();
		for (int i = 0; i < valores.size(); i++) {
			Object objeto = new CampanhaAdsMetricaIntraday();
			System.out.println(" --> ObjetoMap ");
			BeanUtil.setProperties(objeto, (Map<String, ? extends Object>) valores.get(i), true);
			this.CampanhaAdsMetricaIntradays.add((CampanhaAdsMetricaIntraday) objeto);
		}
	}
}

File: loopback-angular6/src/app/shared/sdk/models/CampanhaAdsTeste.ts
/* tslint:disable */
import {
  ProdutoAfiliadoHotmart,
  ModeloCampanhaAdsTeste,
  PalavraChaveCampanhaAdsTeste,
  AnuncioCampanhaAdsTeste,
  ContaGoogle,
  CampanhaAdsMetrica,
  CampanhaAdsMetricaIntraday,
  PaginaVenda
} from '../index';

declare var Object: any;
export interface CampanhaAdsTesteInterface {
  "nome"?: string;
  "prontaParaTeste"?: number;
  "codigoAds"?: string;
  "dataEnvioGoogle"?: Date;
  "custoTotal"?: number;
  "receitaTotal"?: number;
  "dataInicio"?: Date;
  "dataTermino"?: Date;
  "qtdeImpressao"?: number;
  "qtdeClique"?: number;
  "ctr"?: number;
  "nomeAds"?: string;
  "cpc"?: number;
  "ativa"?: number;
  "cpaMax"?: number;
  "cpcMax"?: number;
  "codigoAdsGrupoAnuncio"?: string;
  "codigoAdsCampanha"?: string;
  "primaryStatus"?: string;
  "primaryStatusReasons"?: string;
  "modeloCampanhaAdsTesteId"?: number;
  "produtoAfiliadoHotmartId"?: number;
  "contaGoogleId"?: number;
  "paginaVendaId"?: number;
  "id"?: number;
  produtoAfiliadoHotmart?: ProdutoAfiliadoHotmart;
  modeloCampanhaAdsTeste?: ModeloCampanhaAdsTeste;
  palavraChaveCampanhaAdsTestes?: PalavraChaveCampanhaAdsTeste[];
  anuncioCampanhaAdsTestes?: AnuncioCampanhaAdsTeste[];
  contaGoogle?: ContaGoogle;
  campanhaAdsMetricas?: CampanhaAdsMetrica[];
  campanhaAdsMetricaIntradays?: CampanhaAdsMetricaIntraday[];
  paginaVenda?: PaginaVenda;
}

export class CampanhaAdsTeste implements CampanhaAdsTesteInterface {
  "nome": string;
  "prontaParaTeste": number;
  "codigoAds": string;
  "dataEnvioGoogle": Date;
  "custoTotal": number;
  "receitaTotal": number;
  "dataInicio": Date;
  "dataTermino": Date;
  "qtdeImpressao": number;
  "qtdeClique": number;
  "ctr": number;
  "nomeAds": string;
  "cpc": number;
  "ativa": number;
  "cpaMax": number;
  "cpcMax": number;
  "codigoAdsGrupoAnuncio": string;
  "codigoAdsCampanha": string;
  "primaryStatus": string;
  "primaryStatusReasons": string;
  "modeloCampanhaAdsTesteId": number;
  "produtoAfiliadoHotmartId": number;
  "contaGoogleId": number;
  "paginaVendaId": number;
  "id": number;
  produtoAfiliadoHotmart: ProdutoAfiliadoHotmart;
  modeloCampanhaAdsTeste: ModeloCampanhaAdsTeste;
  palavraChaveCampanhaAdsTestes: PalavraChaveCampanhaAdsTeste[];
  anuncioCampanhaAdsTestes: AnuncioCampanhaAdsTeste[];
  contaGoogle: ContaGoogle;
  campanhaAdsMetricas: CampanhaAdsMetrica[];
  campanhaAdsMetricaIntradays: CampanhaAdsMetricaIntraday[];
  paginaVenda: PaginaVenda;
  constructor(data?: CampanhaAdsTesteInterface) {
    Object.assign(this, data);
  }
  /**
   * The name of the model represented by this $resource,
   * i.e. `CampanhaAdsTeste`.
   */
  public static getModelName() {
    return "CampanhaAdsTeste";
  }
  /**
  * @method factory
  * @author Jonathan Casarrubias
  * @license MIT
  * This method creates an instance of CampanhaAdsTeste for dynamic purposes.
  **/
  public static factory(data: CampanhaAdsTesteInterface): CampanhaAdsTeste{
    return new CampanhaAdsTeste(data);
  }
  /**
  * @method getModelDefinition
  * @author Julien Ledun
  * @license MIT
  * This method returns an object that represents some of the model
  * definitions.
  **/
  public static getModelDefinition() {
    return {
      name: 'CampanhaAdsTeste',
      plural: 'CampanhaAdsTestes',
      path: 'CampanhaAdsTestes',
      idName: 'id',
      properties: {
        "nome": {
          name: 'nome',
          type: 'string'
        },
        "prontaParaTeste": {
          name: 'prontaParaTeste',
          type: 'number'
        },
        "codigoAds": {
          name: 'codigoAds',
          type: 'string'
        },
        "dataEnvioGoogle": {
          name: 'dataEnvioGoogle',
          type: 'Date'
        },
        "custoTotal": {
          name: 'custoTotal',
          type: 'number'
        },
        "receitaTotal": {
          name: 'receitaTotal',
          type: 'number'
        },
        "dataInicio": {
          name: 'dataInicio',
          type: 'Date'
        },
        "dataTermino": {
          name: 'dataTermino',
          type: 'Date'
        },
        "qtdeImpressao": {
          name: 'qtdeImpressao',
          type: 'number'
        },
        "qtdeClique": {
          name: 'qtdeClique',
          type: 'number'
        },
        "ctr": {
          name: 'ctr',
          type: 'number'
        },
        "nomeAds": {
          name: 'nomeAds',
          type: 'string'
        },
        "cpc": {
          name: 'cpc',
          type: 'number'
        },
        "ativa": {
          name: 'ativa',
          type: 'number'
        },
        "cpaMax": {
          name: 'cpaMax',
          type: 'number'
        },
        "cpcMax": {
          name: 'cpcMax',
          type: 'number'
        },
        "codigoAdsGrupoAnuncio": {
          name: 'codigoAdsGrupoAnuncio',
          type: 'string'
        },
        "codigoAdsCampanha": {
          name: 'codigoAdsCampanha',
          type: 'string'
        },
        "primaryStatus": {
          name: 'primaryStatus',
          type: 'string'
        },
        "primaryStatusReasons": {
          name: 'primaryStatusReasons',
          type: 'string'
        },
        "modeloCampanhaAdsTesteId": {
          name: 'modeloCampanhaAdsTesteId',
          type: 'number'
        },
        "produtoAfiliadoHotmartId": {
          name: 'produtoAfiliadoHotmartId',
          type: 'number'
        },
        "contaGoogleId": {
          name: 'contaGoogleId',
          type: 'number'
        },
        "paginaVendaId": {
          name: 'paginaVendaId',
          type: 'number'
        },
        "id": {
          name: 'id',
          type: 'number'
        },
      },
      relations: {
        produtoAfiliadoHotmart: {
          name: 'produtoAfiliadoHotmart',
          type: 'ProdutoAfiliadoHotmart',
          model: 'ProdutoAfiliadoHotmart',
          relationType: 'belongsTo',
                  keyFrom: 'produtoAfiliadoHotmartId',
          keyTo: 'hotmartId'
        },
        modeloCampanhaAdsTeste: {
          name: 'modeloCampanhaAdsTeste',
          type: 'ModeloCampanhaAdsTeste',
          model: 'ModeloCampanhaAdsTeste',
          relationType: 'belongsTo',
                  keyFrom: 'modeloCampanhaAdsTesteId',
          keyTo: 'id'
        },
        palavraChaveCampanhaAdsTestes: {
          name: 'palavraChaveCampanhaAdsTestes',
          type: 'PalavraChaveCampanhaAdsTeste[]',
          model: 'PalavraChaveCampanhaAdsTeste',
          relationType: 'hasMany',
                  keyFrom: 'id',
          keyTo: 'campanhaAdsTesteId'
        },
        anuncioCampanhaAdsTestes: {
          name: 'anuncioCampanhaAdsTestes',
          type: 'AnuncioCampanhaAdsTeste[]',
          model: 'AnuncioCampanhaAdsTeste',
          relationType: 'hasMany',
                  keyFrom: 'id',
          keyTo: 'campanhaAdsTesteId'
        },
        contaGoogle: {
          name: 'contaGoogle',
          type: 'ContaGoogle',
          model: 'ContaGoogle',
          relationType: 'belongsTo',
                  keyFrom: 'contaGoogleId',
          keyTo: 'id'
        },
        campanhaAdsMetricas: {
          name: 'campanhaAdsMetricas',
          type: 'CampanhaAdsMetrica[]',
          model: 'CampanhaAdsMetrica',
          relationType: 'hasMany',
                  keyFrom: 'id',
          keyTo: 'campanhaAdsTesteId'
        },
        campanhaAdsMetricaIntradays: {
          name: 'campanhaAdsMetricaIntradays',
          type: 'CampanhaAdsMetricaIntraday[]',
          model: 'CampanhaAdsMetricaIntraday',
          relationType: 'hasMany',
                  keyFrom: 'id',
          keyTo: 'codigoAds'
        },
        paginaVenda: {
          name: 'paginaVenda',
          type: 'PaginaVenda',
          model: 'PaginaVenda',
          relationType: 'belongsTo',
                  keyFrom: 'paginaVendaId',
          keyTo: 'id'
        },
      }
    }
  }
}
